lib/Auth/utils.php

```diff
@@ -1,124 +1,134 @@
+        if ($headers) {curl_setopt($curl_instance, CURLOPT_HTTPHEADER, $headers);}
+        curl_setopt($curl_instance, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($curl_instance, CURLOPT_USERAGENT, USERAGENT);
+        if ($http_method === "POST") {
+            curl_setopt($curl_instance, CURLOPT_POST, true);
+            curl_setopt($curl_instance, CURLOPT_POSTFIELDS, $params);
+        } elseif ($http_method === "GET") {
+            $params_str = '?';
+            foreach ($params as $key => $value) {
+                $params_str .= $key . "=" . $value . "&";
+            }
+            curl_setopt($curl_instance, CURLOPT_URL, $url . $params_str);
+        }
+        if (!$serverconfig['sslverifyhost']) {
+            curl_setopt($curl_instance, CURLOPT_SSL_VERIFYHOST, 0);
+        }
+        curl_setopt($curl_instance, CURLOPT_SSL_VERIFYPEER, $serverconfig['sslverifypeer']);
+
+        SimpleSAML_Logger::debug("privacyIDEA: " . $http_method . " " . $url);
+        $response = curl_exec($curl_instance);
+        if (!$response) {
+            throw new SimpleSAML_Error_BadRequest(
+                "privacyIDEA: Bad request to PI server: " . curl_error($curl_instance)
+            );
+        }
+        $header_size = curl_getinfo($curl_instance, CURLINFO_HEADER_SIZE);
+        $body = json_decode(substr($response, $header_size));
+        return $body;
+    }
+
+    /**
+     * Null check.
+     *
+     * This function will null-check a value and throw a BadRequest error with a generic response if necessary.
+     *
+     * @param mixed $x Variable to perform the null-check on.
+     * @return mixed The input.
+     * @throws SimpleSAML_Error_BadRequest
+     */
+    public static function nullCheck($x) {
+        if (gettype($x) === "NULL") {
+            throw new SimpleSAML_Error_BadRequest(
+                "privacyIDEA: We were not able to read the response from the PI server");
+        }
+        return $x;
+    }
+
+    /**
+     * With this function you can get the authorization token with a service account.
+     *
+     * The keys serviceAccount and servicePass must be set in the $serverconfig. It can be done in the config or in the
+     * metadata. The service account must have the correct rights. You can edit them in the policies in privacyIDEA
+     *
+     * @param array $serverconfig The whole configuation for the server.
+     * @return String This is the authorization header, which is needed for some API requests.
+     */
+    public static function fetchAuthToken($serverconfig)
+    {
+        assert('array' === gettype($serverconfig));
+
+        return self::nullCheck(
+            @self::curl(
+                array(
+                    "username" => $serverconfig['serviceAccount'],
+                    "password" => $serverconfig['servicePass'],
+                ),
+                null,
+                $serverconfig,
+                "/auth",
+                "POST"
+            )->result->value->token
```

Comment:
``result->value->token `` does not exist, if the authentication fails - e.g. the servicePass is wrong.
In this case a probably strange exception will be raised!?